packages/fiori/src/BarcodeScannerDialog.ts

```diff
@@ -230,19 +235,64 @@ class BarcodeScannerDialog extends UI5Element {
 
 	_decodeFromCamera() {
 		const videoElement = this._getVideoElement();
-		this._codeReader.decodeFromVideoDevice(null, videoElement, (result: Result, err?: Exception) => {
-			this.loading = false;
-			if (result) {
-				this.fireEvent<BarcodeScannerDialogScanSuccessEventDetail>("scan-success",
-					{
-						text: result.getText(),
-						rawBytes: result.getRawBytes(),
-					});
-			}
-			if (err && !(err instanceof NotFoundException)) {
-				this.fireEvent<BarcodeScannerDialogScanErrorEventDetail>("scan-error", { message: err.message });
-			}
-		}).catch((err: Error) => this.fireEvent<BarcodeScannerDialogScanErrorEventDetail>("scan-error", { message: err.message }));
+
+		try {
+			this._codeReader.decodeFromVideoDevice(null, videoElement, (result: Result, err?: Exception) => {
+				this.loading = false;
+
+				if (result) {
+					this._handleDecodeSuccess(result, videoElement);
```

Comment:
Do not pass `videoElement`, get it again from the other function. Only pass the result object.